tray.c

```diff
@@ -182,13 +182,22 @@ CreatePopupMenus()
 
     hMenu = o.groups[0].menu; /* the first group menu is also the root menu */
 
+    /* Save menudata from submenu */
+    UINT_PTR menuData = minfo.dwMenuData;
```

Comment:
Setting this to "the last value of the loop" feels a bit hacky. See below for an alternative approach.